### ISSUE TITLE
add click functionality

### ISSUE THREAD

**Issue by ghost** [finestructure/Hummingbird#26]
I know it seems somewhat counterintuitive to the spirit of this project vs easy-move-resize, but adding a "require click" functionality would be wonderful.

I could use easy-move-resize, but I much prefer hummingbird's ability to have different shortcuts for move and resize (option for move, shift+option for resize) which the original project lacks.

really looking for a best of both worlds!

**Comment by finestructure** [comment]
Hah, we're coming full circle! :)

Originally, HB's functionality was a PR on easy-move-resize to add "hover style" move and resize. However, I didn't manage within a reasonable amount of time to make both modes work well together in @dmarcotte 's code base (not for any fault of Daniel's code, just my lack of time and understanding of the intricacies). Daniel and I agreed that it's best to not even try and merge it in and instead keep a fork. My fork of EMR then became HB.

Leaving aside the feasability for now (I suspect it shouldn't be _too_ difficult to add mouse clicks as another trigger criterion to HB - but then again, I failed at this once before 😆), given the history of the two projects, I wouldn't want to even explore that unless @dmarcotte has confirmed it not being a use case he wants to support.

Have you had a look over in the EMR project to see if there's interest for this use case?

**Comment by ghost** [comment]
Yeah that’s really the next step. I’ll poke around over there and see how he feels about things.

Appreciate the response :)

**Comment by ghost** [comment]
left a comment over on EMR, but project looks largely abandoned.

happy to close out this issue if you don't intend to implement clicks - no reason to leave it kicking about :)

**Comment by finestructure** [comment]
Thanks again for raising the issue - closing this now as part of some housekeeping :)

**Comment by alexandre-g** [comment]
@finestructure 
Hi sorry to bring this back from the dead, but I'm also after the click-button-drag behaviour and upon trying EMR and Hummingbird found that the latter is much smoother for both moving and resizing. Any idea why that is?

**Comment by finestructure** [comment]
Mmm, that's puzzling, because the core functionality is pretty much the same, except I've ported it to Swift. I might have tweaked some parameters a little but haven't personally found them to make much of a difference.

For instance in version 3.1.1 I changed some parameters and in 3.1.3 I reverted them back. If you find these versions to behave differently for you then perhaps there is indeed an effect that I perhaps just didn't see on my Mac.
